src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+            LogInfo.Log($"{nameof(SetSecurityAccessModule)} Write: {ret}", LogLevel.Debug);
+            if (ret < 0) return false;
+            // We don't expect any result, just that the command went well
+            ret = ReadResponse(CommandSet.SAMConfiguration, Span<byte>.Empty);
+            LogInfo.Log($"{nameof(SetSecurityAccessModule)} read: {ret}", LogLevel.Debug);
+            return ret >= 0;
+        }
+
+        private bool CheckVersion()
+        {
+            var ret = WriteCommand(CommandSet.GetFirmwareVersion);
+            LogInfo.Log($"GetFirmwareVersion write command returned: {ret}", LogLevel.Info);
+            if (ret < 0) return false;
+            Span<byte> firmware = stackalloc byte[4];
+            ret = ReadResponse(CommandSet.GetFirmwareVersion, firmware);
+            var ver = firmware.ToArray();
+            if (ret >= 0)
+            {
+                FirmwareVersion = new FirmwareVersion()
+                {
+                    IdentificationCode = firmware[0],
+                    Version = new Version(firmware[1], firmware[2]),
+                    VersionSupported = (VersionSupported)(firmware[3] & 0b0000_0111)
+                };
+            }
+            LogInfo.Log($"GetFirmwareVersion read command returned: {ret} - Bytes {BitConverter.ToString(ver)}", LogLevel.Info);
+            return ret >= 0;
+        }
+
+        /// <summary>
+        /// Get or set the Security Access Module parameters
+        /// </summary>
+        public ParametersFlags ParametersFlags
+        {
+            get { return _parametersFlags; }
+
+            set
+            {
+                if (SetParameters(value))
+                    _parametersFlags = value;
+            }
+        }
+        private bool SetParameters(ParametersFlags parametersFlags)
+        {
+            Span<byte> toSend = stackalloc byte[1] { (byte)parametersFlags };
+            var ret = WriteCommand(CommandSet.SetParameters, toSend);
+            if (ret < 0) return false;
+            ret = ReadResponse(CommandSet.SetParameters, Span<byte>.Empty);
+            LogInfo.Log($"{nameof(SetParameters)}: {ret}", LogLevel.Debug);
+            return ret >= 0;
+        }
+
+        /// <summary>
+        /// List all targets cards in range
+        /// When using this function, you can't determine which target you've read
+        /// So you'll need to use the Decode functions to try to get a card type
+        /// So use this function only with a specific card type. Prefer the AutoPoll function
+        /// As the type identified is returned
+        /// </summary>
+        /// <param name="maxTarget">The maximum number of targets</param>
+        /// <param name="targetBaudRate">The baud rate to use</param>
+        /// <returns>A raw byte array with the data of the targets if any has been identified</returns>
+        public byte[] ListPassiveTarget(MaxTarget maxTarget, TargetBaudRate targetBaudRate)
```

Comment:
I'm not familiar with this API - does targetBaudRate refer to "find me all cards talking with this speed" or rather "use this baud rate to find all cards"? This should be clarified in the comment. Also `byte[]` output is weird - is one card represented by one byte or should this provide some way to enumerate on these bytes (fine to leave this as is for now but please create an issue)

Reply:
I was struggling with the byte[] return. The device returns a byte[] with [0] as the target number. The rest are the information related to the card it found. And depending of the type, the length of the array is not the same. Reason why there are the Decode functions. To be able to find out what is returned if you are looking at diverse cards.
This is a bit different than the AutoPoll function which returne [0] the target number [1] the type of card and then the raw data.
So we can make this evolve, feedback welcome once you'll play with it a bit so we can decide the best way to have this handled. And same apply to the AutoPoll function.
For the baud rate, it's the baudrate which will be used to find the card.